packages/dds/tree/src/feature-libraries/modular-schema/fieldChangeHandler.ts

```diff
@@ -254,9 +258,35 @@ export interface ModularChangeset {
 	 * If undefined then this changeset contains no IDs.
 	 */
 	maxId?: ChangesetLocalId;
+	/**
+	 * The revisions included in this changeset, ordered temporally (oldest to newest).
+	 * Undefined for anonymous changesets.
+	 * Should never be empty.
+	 */
+	readonly revisions?: readonly RevisionInfo[];
 	changes: FieldChangeMap;
 }
 
+/**
+ * A callback that returns the index of the changeset associated with the given RevisionTag among the changesets being
+ * composed or rebased. This index is solely meant to communicate relative ordering, and is only valid within the scope of the
+ * compose or rebase operation.
+ *
+ * During composition, the index reflects the order of the changeset within the overall composed changeset that is
+ * being produced.
+ *
+ * During rebase, the indices of the base changes are all lower than the indices of the change being rebased.
+ * @alpha
+ */
+export type RevisionIndexer = (tag: RevisionTag) => number;
+
+/**
+ * @alpha
+ */
+export interface RevisionInfo {
```

Comment:
You may be wondering why I bothered to wrap the RevisionTag in a struct.
I'm thinking of putting more things in here. Specifically, whether the change is a rollback.
This would make my other PR for the Uninsert support cleaner.